src/boot/efi/stub.c

```diff
@@ -467,27 +567,10 @@ static EFI_STATUS run(EFI_HANDLE image) {
          * measure the additions separately, after the embedded options, but before the smbios ones,
          * so that the order is reversed from "most hardcoded" to "most dynamic". The global addons are
          * loaded first, and the image-specific ones later, for the same reason. */
-        err = cmdline_append_and_measure_addons(
-                        image,
-                        loaded_image,
-                        u"\\loader\\addons",
-                        uname,
-                        &m,
-                        &cmdline);
-        if (err != EFI_SUCCESS)
-                log_error_status(err, "Error loading global addons, ignoring: %m");
+        cmdline_append_and_measure_addons(cmdline_addons_global, &cmdline, &m);
         parameters_measured = parameters_measured < 0 ? m : (parameters_measured && m);
 
-        _cleanup_free_ char16_t *dropin_dir = get_extra_dir(loaded_image->FilePath);
-        err = cmdline_append_and_measure_addons(
-                        image,
-                        loaded_image,
-                        dropin_dir,
-                        uname,
-                        &m,
-                        &cmdline);
-        if (err != EFI_SUCCESS)
-                log_error_status(err, "Error loading UKI-specific addons, ignoring: %m");
+        cmdline_append_and_measure_addons(cmdline_addons_uki, &cmdline, &m);
```

Comment:
hmm, actually it kinda sucks generating multiple measurement records for these measurements. I'd prefer if we could merge that into one measurement. I udnerstand that this is a pre-existing problem, but maybe fix that.

It's particularly bad that we can't distinguish the measurements, i.e.we measure the global and the per-uki cmdlines separately, but we cannot determine afterwards if we have one measurement which of the two it was. Hence we have the distinction that is useless there.

I think we should improve this at least a tiny bit here, and merge the per-uki and global cmdline measurement into one, i.e. merge all additions first, measure them, and then append them to the overall cmdline.

This kinda matters for me for the pcrlock work: i must have a sane way to reproduce the measurements...

(I mean, it's still going to be chaotic enough, as we will have three undiscernable measurements in the worst case: addon cmdline, smbios cmdline and process cmdline).

Reply:
ok, done